specification/compute/resource-manager/Microsoft.Compute/preview/2020-09-30/sharedGallery.json

```diff
@@ -0,0 +1,1040 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "SharedGalleryServiceClient",
+    "description": "Shared Gallery Service Client.",
+    "version": "2020-09-30"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/sharedGalleries": {
+      "get": {
+        "tags": [
+          "SharedGalleries"
+        ],
+        "operationId": "SharedGalleries_List",
+        "description": "List shared galleries by subscription id or tenant id.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationNameParamter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "sharedTo",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The query parameter to decide what shared galleries to fetch when doing listing operations.",
+            "enum": [
+              "tenant"
+            ],
+            "x-ms-enum": {
+              "name": "SharedToValues",
+              "modelAsString": true
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SharedGalleryList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get a gallery.": {
+            "$ref": "./examples/ListSharedGalleries.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/sharedGalleries/{galleryUniqueName}/images": {
```

Comment:
what name is given to a shared gallery to ensure it is unique in the destination?

Reply:
The unique name is in the format of "SubscriptionId - GalleryName", subscriptionId is unqiue guid and gallery name under a subscription is unique, so this can guarantee uniqueness.